src/server.c

```diff
@@ -3726,6 +3732,8 @@ int processCommand(client *c) {
                                (c->cmd->proc == execCommand && (c->mstate.cmd_inv_flags & CMD_STALE));
     int is_denyloading_command = !(c->cmd->flags & CMD_LOADING) ||
                                  (c->cmd->proc == execCommand && (c->mstate.cmd_inv_flags & CMD_LOADING));
+    int is_can_replicate_command = (c->cmd->flags & CMD_CAN_REPLICATE) ||
+                                   (c->cmd->flags & CMD_WRITE) || (c->cmd->proc == execCommand);
```

Comment:
we don't want to consider any exec a write command, only ones that contains a write command.
```suggestion
    int is_can_replicate_command = (c->cmd->flags & (CMD_WRITE | CMD_CAN_REPLICATE)) ||
                           (c->cmd->proc == execCommand && (c->mstate.cmd_flags & (CMD_WRITE | CMD_CAN_REPLICATE)));
```

Reply:
Ok, I'm going to add a few more test cases around this as well.

Reply:
so you mean you're gonna apply my suggestion, right?